packages/recommend-vdom/src/DefaultHeader.tsx

```diff
@@ -5,6 +5,13 @@ import { cx } from './utils';
 
 export function createDefaultHeaderComponent({ createElement }: Renderer) {
   return function DefaultHeader<TObject>(props: ComponentProps<TObject>) {
+    if (
+      (!props.recommendations || props.recommendations.length < 1) &&
+      props.removeHeaderComponent
```

Comment:
Maybe the name is not clear enough as this will only remove the header when there is no results and if the user is not passing a `headerComponent`. WDYT?